go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -343,13 +343,18 @@ func setupOVNNode(node *kapi.Node) error {
 	return nil
 }
 
-func setEncapPort() error {
+func setEncapPort(ctx context.Context) error {
 	systemID, err := util.GetNodeChassisID()
 	if err != nil {
 		return err
 	}
-	uuid, _, err := util.RunOVNSbctl("--data=bare", "--no-heading", "--columns=_uuid", "find", "Encap",
-		fmt.Sprintf("chassis_name=%s", systemID))
+	var uuid string
+	err = wait.PollUntilContextTimeout(ctx, 250*time.Millisecond, 30*time.Second, true,
+		func(ctx context.Context) (bool, error) {
+			uuid, _, err = util.RunOVNSbctl("--data=bare", "--no-heading", "--columns=_uuid", "find", "Encap",
+				fmt.Sprintf("chassis_name=%s", systemID))
+			return len(uuid) != 0, err
+		})
```

Comment:
What happens if this 30s timeout is encountered? Won't we be in a similar situation as we were previously where ovnkube is "stuck". 

Is there a way to guarantee this logic will be eventually consistent (even if it involves crashing/restarting the process) if the timeout is hit?

From my perspective, the issue wasn't so much that we never retried this specific command, but that the process gave up entirely without eventual consistency.

Reply:
Ok, increasing the timeout to a longer value 500s, we do have similar retries logic for node's ic zone and logical switch [checks](https://github.com/ovn-org/ovn-kubernetes/pull/3974/files#diff-2dc6772ca96eb44cc601c3c4daf9d4e6722fcdc1a7050ace708bde0ec9f593cdL714-L764) while starting ovnkube controller. Hope this should be fine now.

Reply:
> Is there a way to guarantee this logic will be eventually consistent (even if it involves crashing/restarting the process) if the timeout is hit?

@pperiyasamy @davidvossel Wasn't this what was happening? The process would crash and be restarted again, even before this fix, to an eventual convergence.

I am fine with this improvement though, as this is something we generally do elsewhere as well to avoid the crash-loop backoff timeouts & delays with something that might take a bit to converge but not that much.

Reply:
@jcaamano no, the NetworkControllerManager is started in parallel and it doesn't attempt to restart upon failure (it may be due to some obvious reason?) i guess that is the reason we do have longer waits (300s) at required places to achieve eventual convergence.

Reply:
@davidvossel 

After looking at this a bit more with @pperiyasamy, our current thinking is that we are not terminating up properly as we should after this error, so there is no crash and there is no eventual convergence after a restart. I have agreed with @pperiyasamy to open a new bug/PR to look at this. 

We will move forward with the solution in this PR however as even if that crash was in place hypothetically, this change would just get things up and running quicker.

Also worth mentioning that setting the encap port is no longer needed, as OVN should be able to pick up from the OVS DB. This is probably here from past times where OVN did not do that. We will keep it for now.